Core/Systems/EcsReactiveSystem.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+
+namespace Secs
+{
+    public abstract class EcsReactiveSystem : IEcsReactiveSystem
+    {
+        protected enum ComponentReactiveState
+        {
+            ComponentAdded,
+            ComponentRemoved,
+            ComponentAddedOrRemoved
+        }
+        
+        private EcsWorld _ecsWorld;
+        private ComponentReactiveState _componentReactiveState;
+        private EcsFilter _filter;
+        private Type _observeType;
+        private bool _activated;
+        
+        internal void Init(EcsWorld ecsWorld)
+        {
+            _ecsWorld = ecsWorld;
+            _componentReactiveState = ObserveOnState();
+            _filter = ObserveFilter(in _ecsWorld);
+            _observeType = ObserveOnType();
+            
+            Activate();
+        }
+        
+        private void OnComponentAddedToEntity(int entityId, Type type) => OnComponentAction(entityId, type);
+        private void OnComponentDeletedFromEntity(int entityId, Type type) => OnComponentAction(entityId, type);
+        
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void OnComponentAction(int entityId, Type type)
+        {
+            if(type != _observeType)
+                return;
+
+            if (_filter == null)
+            {
+                OnExecute(in entityId);
+                return;
+            }
+            
+            if (!_filter.HasEntity(entityId))
+                return;
+
+            OnExecute(in entityId);
+            
+        }
+        
+        protected virtual EcsFilter ObserveFilter(in EcsWorld ecsWorld) => null;
+        protected abstract Type ObserveOnType();
+        protected abstract void OnExecute(in int entityId);
```

Comment:
This will be super costly operation. Firing virtual methods frequently is a huge performance hit. More about it: "https://www.dataorienteddesign.com/dodmain/node4.html#SECTION00000000000000000000